openapi-bal-task-plugin/src/main/java/io/ballerina/openapi/bal/tool/OpenAPICodeGeneratorTool.java

```diff
@@ -136,11 +141,28 @@ public void executeTool(ToolContext toolContext) {
             }
         } catch (BallerinaOpenApiException e) {
             Constants.DiagnosticMessages error = Constants.DiagnosticMessages.PARSER_ERROR;
-            reportDiagnostics(error, e.getMessage(), location);
+            toolContext.reportDiagnostic(createDiagnostics(error, e.getMessage(), location));
         } catch (IOException | FormatterException e) {
             Constants.DiagnosticMessages error = Constants.DiagnosticMessages.ERROR_WHILE_GENERATING_CLIENT;
-            reportDiagnostics(error, e.getMessage(), location);
+            toolContext.reportDiagnostic(createDiagnostics(error, e.getMessage(), location));
+        }
+    }
+
+    /**
+     * This method uses to validate the cache.
+     */
+    private static boolean validateCache(ToolContext toolContext, OASClientConfig clientConfig) throws IOException {
+        Path cachePath = toolContext.cachePath();
+        md5HexOpenAPI = getHashValue(clientConfig, toolContext.targetModule());
+        if (Files.isDirectory(cachePath)) {
```

Comment:
We can reduce the indentation if we do an early return for the negative case here:
```java
        if (!Files.isDirectory(cachePath)) {
            return false;
        }
        // other logic here
```